### PR TITLE
feat(veeam-backup): add offer detail component

### PR DESCRIPTION
**Pull request by varun257**
add offer detail as component
reuse the component
fix styling issues

Signed-off-by: Varun257 <varun257@gmail.com>

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/new/new.module.js [hunk 1]
```diff
@@ -4,13 +4,20 @@ import 'ovh-ui-angular';
...
+import offerDetails from '../components/offer-details';
 import routing from './new.routing';
 import termsOfUse from '../components/termsOfUse';
 
 const moduleName = 'ovhManagerDedicatedCloudBackupNewModule';
 
 angular
-  .module(moduleName, ['oui', 'ovh-api-services', offers, termsOfUse])
+  .module(moduleName, [
+    'oui',
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/new/new.module.js]: Order

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -2,15 +2,7 @@
...
-  "dedicatedCloud_offer_upgrade_retention_points": "{{retention}} points de rétention",
-  "dedicatedCloud_offer_upgrade_full_backups": "{{fullBackups}} sauvegardes complètes",
-  "dedicatedCloud_offer_upgrade_replication": "Réplication hors site",
-  "dedicatedCloud_offer_upgrade_choice_backup_days": "Choix des jours pour chaque sauvegarde",
-  "dedicatedCloud_offer_upgrade_choice_backup_time": "Choix de l’heure de début des sauvegardes",
-  "dedicatedCloud_offer_upgrade_long_term_retention": "11 points de rétention longue durée (GPS rétention)",
-  "dedicatedCloud_offer_upgrade_proxy_backup": "1 backup proxy par hôte",
   "dedicatedCloud_offer_current_offer": "Offre actuelle",
-  "dedicatedCloud_offer_price_per_vm": "/VM"
+  "dedicatedCloud_offer_price_per_vm": "/VM",
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json]:
```suggestion
  "dedicatedCloud_offer_price_per_vm": "{{price}}/VM",
```

or else the translators won't be able to understand what is going on here

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -2,15 +2,7 @@
...
+  "dedicatedCloud_offer_price_from": "À partir de"
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json]:
```suggestion
  "dedicatedCloud_offer_price_from": "À partir de {{date}}"
```
or else the translators won't be able to understand what is going on here

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "dedicatedCloud_offer_upgrade_custom_report": "Rapport de sauvegarde personnalisable",
+  "dedicatedCloud_offer_upgrade_encrypted_backup": "Sauvegarde chiffrée",
+  "dedicatedCloud_offer_upgrade_retention_points": "{{retention}} points de rétention",
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: What is there is only one ? :)

```suggestion
  "dedicatedCloud_offer_upgrade_retention_point_1": "1 point de rétention",
  "dedicatedCloud_offer_upgrade_retention_point_+": "{{retention}} points de rétention",
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "dedicatedCloud_offer_upgrade_full_backups": "{{fullBackups}} sauvegardes complètes",
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: What if there is only one ? :)

```suggestion
  "dedicatedCloud_offer_upgrade_full_backup_1": "1 sauvegarde complète",
  "dedicatedCloud_offer_upgrade_full_backup_+": "{{fullBackups}} sauvegardes complètes",
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import angular from 'angular';
+
+import component from './component';
+
+const moduleName = 'ovhManagerDedicatedCloudBackupOfferDetails';
+
+angular
+  .module(moduleName, [])
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js]: No translations, no ui kit, no routing ?

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/template.html [hunk 7]
```diff
@@ -1,143 +1,64 @@
...
+            </oui-select-picker-description>
+        </div>
+        <oui-select-picker-section>
+            <dedicated-cloud-datacenter-backup-offer-details
+                data-offer="offer"
+            ></dedicated-cloud-datacenter-backup-offer-details>
+        </oui-select-picker-section>
+        <oui-select-picker-footer>
+            <span data-translate="dedicatedCloud_offer_price_from"></span>
+            <dedicated-price
```
Comment (FredericEspiau) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/template.html]: We should use https://github.com/ovh/manager/pull/2214/files#diff-d6bd2bba6df155c89b3913e7a2d11453R1 when it's available, (not in this PR, it's just a note)

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json (hunk 3 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json]: From `"price"`

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json (hunk 2 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/translations/Messages_fr_FR.json]: Price is a component. This is the text after displaying price.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js (hunk 6 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js]: Its more like a static component. So routing is not needed. It is used to display details of the offer inside the offer component.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json (hunk 5 above)
Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: We could use `<ng-pluralize>` but i dont know if there is `localization` support for it :)

Comment (FredericEspiau) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]:
```js
<li
                        data-ng-pluralize
                        data-count="CloudProjectDeleteCtrl.resources.instance"
                        data-when="{
                                    'one': ('cloud_project_delete_resource_warning_one_server' | translate),
                                    'other': ('cloud_project_delete_resource_warning_many_server' | translate : {amountInstance: CloudProjectDeleteCtrl.resources.instance})
                                }"
                    ></li>
```

Here is an example from the code

Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: Yess i got it. But there is no 1 backup or 1 retention. All plans its above 1...Ill add this anyway

Comment (FredericEspiau) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: Hum if you're sure there is none, you don't have to do it imo 😉

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  "dedicatedCloud_offer_upgrade_replication": "Réplication hors site",
+  "dedicatedCloud_offer_upgrade_choice_backup_days": "Choix des jours pour chaque sauvegarde",
+  "dedicatedCloud_offer_upgrade_choice_backup_time": "Choix de l’heure de début des sauvegardes",
+  "dedicatedCloud_offer_upgrade_long_term_retention": "11 points de rétention longue durée (GPS rétention)",
+  "dedicatedCloud_offer_upgrade_proxy_backup": "1 backup proxy par hôte"
+}
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]:
```suggestion
}

```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/template.html [hunk 9]
```diff
@@ -1,143 +1,64 @@
...
+                ></span>
+            </oui-select-picker-description>
+        </div>
+        <oui-select-picker-section>
+            <dedicated-cloud-datacenter-backup-offer-details
+                data-offer="offer"
+            ></dedicated-cloud-datacenter-backup-offer-details>
+        </oui-select-picker-section>
+        <oui-select-picker-footer>
+            <span data-translate="dedicatedCloud_offer_price_from"></span>
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offers/template.html]:
```suggestion
            <span data-translate="dedicatedCloud_datacenter_backup_offer_price_from"></span>
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json [hunk 10]
```diff
@@ -0,0 +1,11 @@
+{
+  "dedicatedCloud_offer_upgrade_custom_report": "Rapport de sauvegarde personnalisable",
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/translations/Messages_fr_FR.json]: name must start with "dedicatedCloud_datacenter_backup"
```suggestion
  "dedicatedCloud_datacenter_backup_offer_upgrade_custom_report": "Rapport de sauvegarde personnalisable",
```

make these changes for all translation names

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import angular from 'angular';
+
+import component from './component';
+
+const moduleName = 'ovhManagerDedicatedCloudBackupOfferDetails';
+
+angular
+  .module(moduleName, ['oui'])
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js]: `ovh-ui-angular` is not imported, neither are `angular-translate` and `ngTranslateAsyncLoader`

Comment (varun257) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/components/offer-details/index.js]: Have added this

**Pull request merged**